internal/build/phase_config_provider.go

```diff
@@ -122,14 +123,34 @@ func WithBinds(binds ...string) PhaseConfigProviderOperation {
 	}
 }
 
-func WithDaemonAccess() PhaseConfigProviderOperation {
+func WithDaemonAccess(dockerHost string) PhaseConfigProviderOperation {
 	return func(provider *PhaseConfigProvider) {
 		WithRoot()(provider)
-		bind := "/var/run/docker.sock:/var/run/docker.sock"
-		if provider.os == "windows" {
-			bind = `\\.\pipe\docker_engine:\\.\pipe\docker_engine`
+		if dockerHost == "inherit" {
+			dockerHost = os.Getenv("DOCKER_HOST")
+		}
+		var bind string
+		if dockerHost == "" {
+			bind = "/var/run/docker.sock:/var/run/docker.sock"
+			if provider.os == "windows" {
+				bind = `\\.\pipe\docker_engine:\\.\pipe\docker_engine`
+			}
+		} else {
+			switch {
+			case strings.HasPrefix(dockerHost, "tcp://"):
+				provider.ctrConf.Env = append(provider.ctrConf.Env, fmt.Sprintf(`DOCKER_HOST=%s`, dockerHost))
+			case strings.HasPrefix(dockerHost, "unix://"):
+				bind = fmt.Sprintf("%s:/var/run/docker.sock", strings.TrimPrefix(dockerHost, "unix://"))
+			case strings.HasPrefix(dockerHost, "npipe://") || strings.HasPrefix(dockerHost, `npipe:\\`):
+				sub := ([]rune(dockerHost))[8:]
```

Comment:
Yes, it removes first eight characters. I might as well use `[]byte` instead of `[]rune` since all the character at beginning are one byte wide. Or maybe I could write
```go
([]rune(dockerHost))[len(`npipe://`):]
```